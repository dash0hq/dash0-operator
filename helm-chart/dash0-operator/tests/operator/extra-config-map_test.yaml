suite: test extra config map
templates:
  - operator/extra-config-map.yaml
tests:
  - it: extra config map should match snapshot
    asserts:
      - matchSnapshot: {}

  - it: should render custom configuration values
    set:
      operator:
        collectors:
          daemonSetCollectorContainerResources:
            limits:
              memory: 101Mi
              cpu: 201m
            gomemlimit: 301MiB
            requests:
              memory: 401Mi
              cpu: 501m
          daemonSetConfigurationReloaderContainerResources:
            limits:
              memory: 102Mi
              cpu: 202m
            gomemlimit: 302MiB
            requests:
              memory: 402Mi
              cpu: 502m
          daemonSetFileLogOffsetSyncContainerResources:
            limits:
              memory: 103Mi
              cpu: 203m
            gomemlimit: 303MiB
            requests:
              memory: 403Mi
              cpu: 503m
          filelogOffsetSyncStorageVolume:
            name: offset-storage
            persistentVolumeClaim:
              claimName: offset-storage-claim
          deploymentCollectorContainerResources:
            limits:
              memory: 104Mi
              cpu: 204m
            gomemlimit: 304MiB
            requests:
              memory: 404Mi
              cpu: 504m
          deploymentConfigurationReloaderContainerResources:
            limits:
              memory: 105Mi
              cpu: 205m
            gomemlimit: 305MiB
            requests:
              memory: 405Mi
              cpu: 505m
          daemonSetTolerations:
            - key: key1
              operator: Equal
              value: value1
              effect: NoSchedule
            - key: key2
              operator: Exists
              effect: NoSchedule
    asserts:
      - equal:
          path: data['extra.yaml']
          value:  |-
            collectorDaemonSetCollectorContainerResources:
              gomemlimit: 301MiB
              limits:
                cpu: 201m
                memory: 101Mi
              requests:
                cpu: 501m
                memory: 401Mi
            collectorDaemonSetConfigurationReloaderContainerResources:
              gomemlimit: 302MiB
              limits:
                cpu: 202m
                memory: 102Mi
              requests:
                cpu: 502m
                memory: 402Mi
            collectorDaemonSetFileLogOffsetSyncContainerResources:
              gomemlimit: 303MiB
              limits:
                cpu: 203m
                memory: 103Mi
              requests:
                cpu: 503m
                memory: 403Mi
            collectorFilelogOffsetStorageVolume:
              name: offset-storage
              persistentVolumeClaim:
                claimName: offset-storage-claim
            collectorDeploymentCollectorContainerResources:
              gomemlimit: 304MiB
              limits:
                cpu: 204m
                memory: 104Mi
              requests:
                cpu: 504m
                memory: 404Mi
            collectorDeploymentConfigurationReloaderContainerResources:
              gomemlimit: 305MiB
              limits:
                cpu: 205m
                memory: 105Mi
              requests:
                cpu: 505m
                memory: 405Mi
            daemonSetTolerations:
              - effect: NoSchedule
                key: key1
                operator: Equal
                value: value1
              - effect: NoSchedule
                key: key2
                operator: Exists

  # This test also verifies that the collectorFilelogOffsetStorageVolume is not rendered if
  # operator.collectors.filelogOffsetSyncStorageVolume isn't set as a value.
  - it: should render custom resource settings from legacy paths
    set:
      operator:
        collectorDaemonSetCollectorContainerResources:
          limits:
            memory: 101Mi
            cpu: 201m
          gomemlimit: 301MiB
          requests:
            memory: 401Mi
            cpu: 501m
        collectorDaemonSetConfigurationReloaderContainerResources:
          limits:
            memory: 102Mi
            cpu: 202m
          gomemlimit: 302MiB
          requests:
            memory: 402Mi
            cpu: 502m
        collectorDaemonSetFileLogOffsetSyncContainerResources:
          limits:
            memory: 103Mi
            cpu: 203m
          gomemlimit: 303MiB
          requests:
            memory: 403Mi
            cpu: 503m
        collectorDeploymentCollectorContainerResources:
          limits:
            memory: 104Mi
            cpu: 204m
          gomemlimit: 304MiB
          requests:
            memory: 404Mi
            cpu: 504m
        collectorDeploymentConfigurationReloaderContainerResources:
          limits:
            memory: 105Mi
            cpu: 205m
          gomemlimit: 305MiB
          requests:
            memory: 405Mi
            cpu: 505m
    asserts:
      - equal:
          path: data['extra.yaml']
          value:  |-
            collectorDaemonSetCollectorContainerResources:
              gomemlimit: 301MiB
              limits:
                cpu: 201m
                memory: 101Mi
              requests:
                cpu: 501m
                memory: 401Mi
            collectorDaemonSetConfigurationReloaderContainerResources:
              gomemlimit: 302MiB
              limits:
                cpu: 202m
                memory: 102Mi
              requests:
                cpu: 502m
                memory: 402Mi
            collectorDaemonSetFileLogOffsetSyncContainerResources:
              gomemlimit: 303MiB
              limits:
                cpu: 203m
                memory: 103Mi
              requests:
                cpu: 503m
                memory: 403Mi
            collectorDeploymentCollectorContainerResources:
              gomemlimit: 304MiB
              limits:
                cpu: 204m
                memory: 104Mi
              requests:
                cpu: 504m
                memory: 404Mi
            collectorDeploymentConfigurationReloaderContainerResources:
              gomemlimit: 305MiB
              limits:
                cpu: 205m
                memory: 105Mi
              requests:
                cpu: 505m
                memory: 405Mi
            daemonSetTolerations:
              []