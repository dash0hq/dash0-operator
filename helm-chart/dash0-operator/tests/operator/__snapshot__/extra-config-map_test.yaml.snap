extra config map should match snapshot:
  1: |
    apiVersion: v1
    data:
      extra.yaml: |-
        initContainerResources:
          {}
        collectorDaemonSetCollectorContainerResources:
          gomemlimit: 400MiB
          limits:
            memory: 500Mi
          requests:
            memory: 500Mi
        collectorDaemonSetConfigurationReloaderContainerResources:
          gomemlimit: 18MiB
          limits:
            memory: 24Mi
          requests:
            memory: 12Mi
        collectorDaemonSetFileLogOffsetSyncContainerResources:
          gomemlimit: 24MiB
          limits:
            memory: 32Mi
          requests:
            memory: 32Mi
        daemonSetTolerations:
          []
        collectorDeploymentCollectorContainerResources:
          gomemlimit: 400MiB
          limits:
            memory: 500Mi
          requests:
            memory: 500Mi
        collectorDeploymentConfigurationReloaderContainerResources:
          gomemlimit: 18MiB
          limits:
            memory: 24Mi
          requests:
            memory: 12Mi
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: extra-config-map
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: dash0-operator
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-extra-config
      namespace: NAMESPACE
